addons/account/models/account_analytic_account.py

```diff
@@ -18,26 +18,40 @@ class AccountAnalyticAccount(models.Model):
     @api.depends('line_ids')
     def _compute_invoice_count(self):
         sale_types = self.env['account.move'].get_sale_types(include_receipts=True)
-        domain = [
-            ('move_line_id.move_id.move_type', 'in', sale_types),
-            ('account_id', 'in', self.ids)
-        ]
-        groups = self.env['account.analytic.line']._read_group(domain, ['move_line_id.move_id:count_distinct'], ['account_id'])
-        moves_count_mapping = dict((g['account_id'][0], g['account_id_count']) for g in groups)
```

Comment:
> The smart buttons were computing the number of aml instead of the number of invoices/bills with the account.

Wasn't that just because you are not reading the right field from the `_read_group`? :thinking:

Reply:
I don't know, seems to me like it would be the right field no? But I'm not sure if `_read_group` can do count_distinct on related field. Maybe I'm wrong, and can do differently with this function, but I can't see how it was wrong.

But even then, shouldn't we do the search on the `analytic_distribution` of `account_move_line`? If not, then you could see on the button 10 invoices, but when you click on it, see none, because you created manually analytic lines related to invoices.

We could do it the other way by showing all the invoices who have analytic lines, but then it would be weird: you click on an invoice, you don't see your analytic account in the distribution, which would also be weird.